Beginner_Projects/E-Commerce/e-commerce.py

```
@@ -0,0 +1,172 @@
+    
+    def customer_counts(self):
+                value=(self.orders['customer_id']).value_counts().sort_index()
+                return f" {value} "
+    
+    def Customer_Lifetime_Value(self):
+        merge=self.calculate_revenue()
+        CLV=merge.groupby('customer_id')['revenue'].sum()
+        return CLV
+    
+    def purchase_date(self): 
+        date=self.orders.groupby('customer_id')['order_date'].agg(first_purchase='min',last_purchase='max')
```

Ruff (0.12.2)

Redefinition of unused `date` from line 3

(F811)